staging/src/k8s.io/apiserver/pkg/server/options/etcd.go

```diff
@@ -375,6 +375,39 @@ func (s *EtcdOptions) addEtcdHealthEndpoint(c *server.Config) error {
 		return readyCheck()
 	}))
 
+	if s.EtcdServersOverrides != nil {
+		if err := s.addOverrideEtcdHealthEndpoint(c); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (s *EtcdOptions) addOverrideEtcdHealthEndpoint(c *server.Config) error {
+	sc := s.StorageConfig.DeepCopy()
+	for _, override := range s.EtcdServersOverrides {
+		tokens := strings.Split(override, "#")
+		servers := strings.Split(tokens[1], ";")
+		sc.Transport.ServerList = servers
```

Comment:
if multiple overrides are specified (`["apps/deployments#d1.example.com;d2.example.com","apps/replicasets#r1.example.com;r2.example.com"]`), this loop means only the last one is healthchecked

also, if multiple resources are overridden to point to the same set of etcds  (`["apps/deployments#s1.example.com;s2.example.com","apps/replicasets#s1.example.com;s2.example.com"]`), their health/readiness should be aggregated

Reply:
Added a set to de-dup.

Reply:
It seems the latter half of the comment is still not addressed.
The 2 scenarios that Liggitt described should be covered by test.

Reply:
> It seems the latter half of the comment is still not addressed. The 2 scenarios that Liggitt described should be covered by test.

Miss that. 
Good catch, updated.